packages/in_app_purchase/lib/src/in_app_purchase_connection/google_play_connection.dart

```diff
@@ -34,6 +34,51 @@ class GooglePlayConnection
     return _billingClient.isReady();
   }
 
+  @override
+  Future<QueryPastPurchaseResponse> queryPastPurchases(
+      {String applicationUserName}) async {
+    List<PurchasesResultWrapper> responses = await Future.wait([
+      _billingClient.queryPurchaseHistory(SkuType.inapp),
+      _billingClient.queryPurchaseHistory(SkuType.subs)
+    ]);
+
+    BillingResponse errorInApp = responses.first.responseCode;
+    BillingResponse errorSubs = responses.last.responseCode;
+    String errorMessage = null;
+    if (errorInApp != BillingResponse.ok) {
+      errorMessage = errorInApp.toString();
+    }
+    if (errorSubs != BillingResponse.ok && errorSubs != errorInApp) {
+      errorMessage += ', ${errorSubs.toString()}';
+    }
```

Comment:
subjective: I think this would be a little cleaner if it was just filtering over the array in one statement instead.

```dart
String errorMessage = responses
  .filter((Response response) => response.responseCode != BillingResponse.ok)
  .map((Response response) => response.responseCode.toString())
  .toList()
  .join(', ');
```